Natural transformation for free freer

xhtmlboi opened the pull request:
For being consistent with `Free_applicative`, `Free_selective` and `Freer_selective`.

gr-im commented:
The natural transformation for Freer seems wrong to me. In `Free`, the continuation is provided explicitly in the type, this is not the case in Freer. So how to provide a natural transformation to `Identity` (for example) when the continuation is implicitly defined.

gr-im commented:
Could you give an example of "resuming" computation for freer?

xhtmlboi commented on lib/preface_make/free_monad.ml:
```diff
@@ -7,6 +7,16 @@ module Via_map (F : Preface_specs.Functor.CORE) = struct
+    let rec run transformation = function
+      | Return a -> Monad.return a
+      | Bind g ->
+        Monad.bind (fun x -> run transformation x) (transformation.transform g)
```
Done! (No particular reason, thanks)

xhtmlboi commented:
> The natural transformation for Freer seems wrong to me. In `Free`, the continuation is provided explicitly in the type, this is not the case in Freer. So how to provide a natural transformation to `Identity` (for example) when the continuation is implicitly defined.

You were absolutely right, I modified the transformation on the Freer and implemented your example in test. Thank you for that! I was probably a bit fast in my implementation :)

gr-im reviewed (approved):
Thanks, It is better (IMHO)
